src/runner/fixture-hook-controller.ts

```diff
@@ -24,11 +26,13 @@ export default class FixtureHookController {
     private static _ensureFixtureMapItem (fixtureMap: Map<Fixture, FixtureState>, fixture: Fixture): void {
         if (!fixtureMap.has(fixture)) {
             const item = {
-                started:                  false,
-                runningFixtureBeforeHook: false,
-                fixtureBeforeHookErr:     null,
-                pendingTestRunCount:      0,
-                fixtureCtx:               Object.create(null),
+                started:                        false,
+                runningFixtureBeforeHook:       false,
+                fixtureBeforeHookErr:           null,
+                pendingTestRunCount:            0,
+                fixtureCtx:                     Object.create(null),
+                fixtureGlobalBeforeHookPromise: Promise.resolve(true),
+                fixtureBeforeHookPromise:       Promise.resolve(true),
```

Comment:
I believe that one promise is enough for this purpose

Reply:
I am not sure about this. Here is why:
When the first item runs its globalBeforeHook, we store its promise in our single field.
Then, this first item will wait until its resolved.
Concurrent items also will wait until its resolved.
However, when the first item runs its localBeforeHook and stores its promise,
Concurrent items will not wait for anything, they have already waited for the single promise field and they will continue execution.

Reply:
you can make single promise for both hooks